extension/lib/doofinder/Client.js

```diff
@@ -27,25 +30,139 @@ class Client {
       },
       json: true
     })
+
     if (response.statusCode >= 400) {
       this.log.error(`Doofinder error code ${response.statusCode} in response`, response.body)
     }
 
-    return response
+    return response.body
   }
 
   /**
-   * @param {SearchRequest} searchPhrase
-   * @returns {Array}
+   * @param {String} query
+   * @param {Object} filters
+   * @param {Number} offset
+   * @param {Number} limit
+   * @param {Object} sort
+   *
+   * @return {Object}
    */
-  async searchSuggestions (searchPhrase) {
-    const titles = []
-    const response = await this.request({query: searchPhrase.query}, 'suggest')
-    for (const result of response.body.results) {
-      titles.push(result.term.charAt(0).toUpperCase() + result.term.slice(1))
+  async paginatedRequest (query, filters, offset, limit, sort) {
+    const firstPage = Math.floor(offset / this.resulsPerPage) + 1
+    const lastPage = Math.ceil((offset + limit) / this.resulsPerPage)
+    const skipCount = offset % this.resulsPerPage
+    let results = []
+    let totalProductCount = 0
+
+    for (let currentPage = firstPage; currentPage <= lastPage; currentPage++) {
+      const response = await this.request({query, filter: filters, page: currentPage, sort})
+      totalProductCount = response.total
+      results = results.concat(response.results)
     }
 
-    return titles
+    return {
+      results: results.slice(skipCount, limit + skipCount),
+      totalProductCount
+    }
+  }
+
+  /**
+   * @param {String} query
+   *
+   * @returns {Object}
+   */
+  async getSearchSuggestions (query) {
+    const response = await this.request({query}, 'suggest')
+
+    return {
+      suggestions: response.results.map(
+        result => result.term.charAt(0).toUpperCase() + result.term.slice(1)
+      )
+    }
+  }
+
+  /**
+   * @param {PipelineInput} param0
+   *
+   * @return {Object}
+   */
+  async searchProducts ({searchPhrase, filters, offset, limit, sort}) {
```

Comment:
`offset` and `limit` are optional for pipeline input. We need fallback values. 
Otherwise you have `NaN` at L:51 `Math.floor`